### ISSUE TITLE
Local registration to ensure sublanguages to be "statically linked"

### ISSUE THREAD

**Issue by goto40** [textX/textX#180]
**Use case**: a language L1 uses a language L2. If  'another' language "L2" is installed it could break the language L1.

Local registration solves this problem. 

**Possible solution** (discussion):
 * We could allow to pass an optional (new) "Registration-Object" with LanguageDesc objects inside when creating a meta model.
 * This local "Registration-Object" would override global registration in case of conflicts... We could encapsulate the registration.py-logic in that "Registration Object". One instance will exist globally (to implement the existing global registration API).
 * The existing global registration API is untouched.
 * The local lookup/registration object could be accessible via _tx_attribute for, e.g., editors.
 * Local registration could be automatically forwarded to global registration, if no conflicts exist.

**Comment by danixeee** [comment]
There are at least two solutions:

- to have _different_ _virtualenvs_ and avoid language collision
- to allow languages with the same name/extension, but to implement local registration

right?

If we choose the first solution then I have to change `textX-LS` architecture. Currently when extension is installed, _one_ _virtualenv_ is created in it and all necessary requirements are installed there. Then, when we install a new language, it will be installed in the _same_ _virtualenv_. So, it would be good to discuss possible solutions for this in the near future.
(I can work on several generators in the meantime).

**Comment by goto40** [comment]
Virtualenvs are an orthogonal concept to me: even with virtual envs you can get collisions, since you cannot control (1) dependencies installed inside the venv and (2) cannot control the libraries you 'inherit', when creating a venv. Am I right? 

But you can surely find more than one solution to the problem described above.

**Comment by goto40** [comment]
@danixeee, I think this issue/ticket is of low priority, since we already see solutions, which are easy to integrate. Local registration is surely not design critical, at the moment.

For you excellent work, I think, you can focus on global registration. Once we have collected experience with the editor support, we can go further here...

**Comment by igordejanovic** [comment]
@goto40 Exactly. It's better to wait some time until we gather more experience with multiple languages used inside textX-LS. We'll be able to make a better decision then.
